servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ClientStrategyInfluencerChainBuilder.java

```diff
@@ -67,50 +68,70 @@ void add(HttpLoadBalancerFactory<?> lb) {
     }
 
     private void add(String purpose, ExecutionStrategyInfluencer<?> influencer, HttpExecutionStrategy strategy) {
+        assert offloadNever() != strategy && defaultStrategy() != strategy :
+                "illegal " + purpose + " required strategy (" + strategy + ") for " + influencer;
         if (offloadNever() == strategy) {
             LOGGER.warn("Ignoring illegal {} required strategy ({}) for {}", purpose, strategy, influencer);
             strategy = offloadNone();
         }
-        if (strategy.hasOffloads()) {
-            clientChain = null != clientChain ? clientChain.merge(strategy) : strategy;
+        if (defaultStrategy() == strategy) {
+            LOGGER.warn("Ignoring illegal {} required strategy ({}) for {}", purpose, strategy, influencer);
```

Comment:
Consider something like:
```java
LOGGER.warn("{}#requireOffloads() returns defaultStrategy(), which is unexpected. offloadAll() should be used instead. Making automatic adjustment, consider updating the filter.");
```
Feel free to reword. The main idea is to let users know what was done for them at runtime and what they have to do to avoid this warning.